lib/services.js

```diff
@@ -191,6 +207,26 @@ export default {
                  });
     },
 
+   /**
+    * Check that hashedStream.completedHash matches header contentMd5.
+    * @param {object} request - request linked to the stream containine the data
+    * @param {string} hashedStream - hased stream once completed
+    * {key, dataStoreName (mem/file)}
+    * @param {RequestLogger} log - the current request logger
+    * @return {object} errors
+    */
+    checkHashMatchMD5(request, hashedStream, log) {
+        if (request.contentMD5 && hashedStream.completedHash &&
```

Comment:
3) I would send into this function the request.contentMD5 as an argument and the hashedStream.completedHash as an argument (it just needs these properties).  I would check to make sure hashedStream.completedHash exists before calling this function.